### ISSUE TITLE
Can't locate default serializer in JS for classes with named companion object

### ISSUE THREAD

**Issue by Whathecode** [Kotlin/kotlinx.serialization#226]
When a class contains a named companion object, the JS runtime can't locate the serializer for the class, whereas the JVM runtime can.

```
    @Serializable
    internal class ObjectWithNamedCompanion
    {
        companion object Factory
        {
            fun theWorld(): Int { return 42 }
        }
    }

    @Test
    fun test()
    {
        val o = ObjectWithNamedCompanion()
        val json = JSON.stringify( o )
    }
```

> SerializationException: Can't locate default serializer for class ObjectWithNamedCompanion
>       at serializer_5 (build\node_modules\kotlinx-serialization-runtime-js.js:7731:13)

This works when `companion object Factory` is changed to `companion object`.

Before, [this also did not work for the JVM runtime](https://github.com/Kotlin/kotlinx.serialization/issues/95), based on which [I created the following PR](https://github.com/Kotlin/kotlinx.serialization/pull/130). I don't recall any more specifics at the moment.

**Comment by Whathecode** [comment]
The culprit is in [JS Serialization.kt](https://github.com/Kotlin/kotlinx.serialization/blob/master/runtime/js/src/main/kotlin/kotlinx/serialization/Serialization.kt).

```
@Suppress("UNCHECKED_CAST")
actual fun <T: Any> KClass<T>.serializer(): KSerializer<T> = this.js.asDynamic().Companion?.serializer() as? KSerializer<T>
        ?: throw SerializationException("Can't locate default serializer for $this")
```

Specifically, `asDynamic().Companion?`. `Companion` here is named after the passed companion object name when it is specified.

I played around a bit with trying to access the members of `JsClass` at runtime to see whether I could iterate over all members and look for a `serializer()` method. However, it seems more magic is going on here, as `JsClass` returns `$metadata$`, with a few other members in it (`kind`, `simpleName`, ...), but no `Companion` or the custom named one. This must have something to do with how dynamic calls are dispatched and is beyond what I can investigate at this point. Hope it helps!

**Comment by sandwwraith** [comment]
Reflection possibilities are very limited on Kotlin/JS, so I doubt that it is even can be fixed. Iterating over dynamic object looking for `.serializer` is promising, but it probably can give false-positives.

In general, some intrinsic can be implemented in compiler to replace this call with retrieving actual serializer.

**Comment by qwwdfsad** [comment]
Fixed in the 0.20.0-1.4.0-rc-95
